### PR TITLE
Convert setup

### PR DESCRIPTION
**Pull request by lastmjs**
### What was wrong?

Related to Issue #962 

### How was it fixed?

I got rid of the requirements .txt files, updated setup.py, tox.ini, and Dockerfile

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by lastmjs** [comment]
If all of my changes work, I believe I'll still need to update some docs.

**Comment by lastmjs** [comment]
Something simple is wrong in `tox.ini`, I'm not sure the format based on https://github.com/ethereum/py-evm/blob/master/tox.ini#L45

**Comment by dylanjw** [comment]
The circleci error is connected to what is being discussed in #794 

You are getting errors because pyethereum depends on rlp<1, and rlp>=1 is being installed.  This PR exposes that issue by installing eth-testrpc as part of of the `dev` extra_requires.

File: setup.py [hunk 1]
```diff
@@ -5,6 +5,57 @@
...
+        "pytest-xdist==1.*",
+        "tox>=1.8.0",
+        "tqdm",
+        "when-changed"
+    ]
+}
+
+extras_require['dev'] = (
+    extras_require['tester'] +
+    extras_require['testrpc'] +
```
Comment (dylanjw) [inline setup.py]: Removing this line should fix the failing tests.

**Comment by lastmjs** [comment]
Removing the installation of eth-testrpc did not work, and I'm not sure now why pytest isn't being installed correctly

**Comment by lastmjs** [comment]
Everything seems to be working fine now, I'm really not sure what the imports being incorrectly sorted problem is with the linter.

**Comment by dylanjw** [comment]
Move the `dev` dependencies back to deps, instead of "extra".  The lint tox job needs web3 and its dependencies installed for isort to understand proper import ordering.

**Pull request merged**